src/data/polynomial/degree/lemmas.lean

```diff
@@ -139,6 +139,39 @@ lemma nat_degree_lt_coeff_mul (h : p.nat_degree + q.nat_degree < m + n) :
   (p * q).coeff (m + n) = 0 :=
 coeff_eq_zero_of_nat_degree_lt (nat_degree_mul_le.trans_lt h)
 
+lemma coeff_mul_of_nat_degree_le (pm : p.nat_degree ≤ m) (qn : q.nat_degree ≤ n) :
+  (p * q).coeff (m + n) = p.coeff m * q.coeff n :=
+begin
+  rcases eq_or_lt_of_le pm with rfl | hm;
+  rcases eq_or_lt_of_le qn with rfl | hn,
+  { exact nat_degree_add_coeff_mul _ _ },
+  { rw [coeff_eq_zero_of_nat_degree_lt hn, mul_zero],
+    exact nat_degree_lt_coeff_mul (add_lt_add_left hn _) },
+  { rw [coeff_eq_zero_of_nat_degree_lt hm, zero_mul],
+    exact nat_degree_lt_coeff_mul (add_lt_add_right hm _) },
+  { rw [coeff_eq_zero_of_nat_degree_lt hn, mul_zero],
+    exact nat_degree_lt_coeff_mul (add_lt_add hm hn) },
+end
+
+lemma coeff_pow_of_nat_degree_le (pn : p.nat_degree ≤ n) :
+  (p ^ m).coeff (n * m) = (p.coeff n) ^ m :=
+begin
+  induction m with m hm,
+  { simp },
+  { rw [pow_succ', pow_succ', ← hm, nat.mul_succ, coeff_mul_of_nat_degree_le _ pn],
+    refine nat_degree_pow_le.trans (le_trans _ (mul_comm _ _).le),
+    exact mul_le_mul_of_nonneg_left pn m.zero_le }
+end
+
+lemma coeff_add_succ_eq_left_of_le (qn : q.nat_degree ≤ n) :
+  (p + q).coeff (n + 1) = p.coeff (n + 1) :=
```

Comment:
Is there any reason why we should not replace this with the stronger statement:
```suggestion
lemma coeff_add_eq_left_of_lt (qn : q.nat_degree < n) :
  (p + q).coeff n = p.coeff n :=
```
and likewise the other `succ` lemmas below?

Reply:
My main motivation is that these lemmas are extracted from `tactic.interactive.compute_degree` and so I particularly care about the exact form that they have.  I can glue the tactic around this other version, though I am also not so sure that it is actually stronger: `q.nat_degree` can be strictly smaller than `n` only if `n` is non-zero anyway.  Thus, it seems that this is just shuffling around where you prove that you are computing a coefficient of degree at least one.  This newer version would probably be literally stronger for `degree` (where you can go below `0`), instead of `nat_degree`.

Reply:
To give a little bit of extra context: I use the tactic `compute_degree_le` to deal with side-goals, so I prefer to leave goals of the form `nat_degree _ \le _`.  If there is one with a strict inequality, it is a trivial fix to precompose with the appropriate `nat.succ` lemma that will convert it to the correct goal.

Reply:
Thanks for the correction; I agree that my suggested form is not strictly stronger.

However I still prefer the versions that I have proposed because the _type_ `(p + q).coeff n = p.coeff n` will match more generally than `(p + q).coeff (n + 1) = p.coeff (n + 1)`, and so should be slightly more useful.

Do you think you could achieve your aims with lemmas stated as I have suggested?

Reply:
Actually, I think that an even better fix would be for `compute_degree_le` to also handle goals of the form `f.nat_degree < d`!  I prefer this fix, rather than working around these lemmas.